Orange/tests/test_data_util.py

```diff
@@ -16,3 +17,30 @@ def test_one_hot(self):
                                          [0, 1, 0],
                                          [0, 0, 1],
                                          [0, 1, 0]])
+
+
+class PlusCommon(SharedComputeValue):
+
+    def compute(self, data, shared_data):
+        return data.X[:,0] + shared_data
+
+
+class TestSharedComputeValue(unittest.TestCase):
+
+    def test_compat_compute_value(self):
+        data = Orange.data.Table("iris")
+        obj = PlusCommon()
+        obj.compute_shared = lambda data: 1.
+        res = obj(data)
+        obj = lambda data: data.X[:,0] + 1.
+        res2 = obj(data)
+        np.testing.assert_equal(res, res2)
+
+    def test_with_row_indices(self):
+        obj = PlusCommon()
+        obj.compute_shared = lambda data: 1.
+        data = Orange.data.Table("iris")
+        domain = Orange.data.Domain([Orange.data.ContinuousVariable("cv", compute_value=obj)])
+        data1 = Orange.data.Table.from_table(domain, data)[:10]
+        data2 = Orange.data.Table.from_table(domain, data, range(10))
+        np.testing.assert_equal(data1.X, data2.X)
```

Comment:
Could you add another test that checks that shared computation is only executed once? You can use [mocks](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_count) as your shared computation functions and then check their call_count.